gspread/worksheet.py

```diff
@@ -1062,18 +1079,25 @@ def update(
                 ValueInputOption.raw if raw is True else ValueInputOption.user_entered
             )
 
-        params = {
-            "valueInputOption": value_input_option,
-            "includeValuesInResponse": include_values_in_response,
-            "responseValueRenderOption": response_value_render_option,
-            "responseDateTimeRenderOption": response_date_time_render_option,
-        }
+        params = extract_enum_values(
+            {
+                "valueInputOption": value_input_option,
+                "includeValuesInResponse": include_values_in_response,
+                "responseValueRenderOption": response_value_render_option,
+                "responseDateTimeRenderOption": response_date_time_render_option,
+            }
+        )
+
+        # get actual value only if not None
+        major_dimension_value = (
+            major_dimension.value if major_dimension is not None else None
```

Comment:
why not use extract_enum_values here?

Reply:
True I could, I will

Reply:
I remember why I did not use the function here:
- the function `extract_enum_values` only takes dict as argument
- So I need to create a dict just to pass the single value
- then get the resulting dict
- if a value put it, if no value put `None`

I felt like this was easier to put